### PR TITLE
DEV_Petrov

### PR DESCRIPTION
**Pull request by grom4er**

### CODE REVIEW

File: src/main/java/MyRunnable.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    private Counter counter;
+
+    public MyRunnable(Counter counter) {
+        this.counter = counter;
+    }
+
+    @Override
+    public void run() {
+        logger.info("MyRunnable thread started:" + System.lineSeparator());
+        for (int i = 0; i < 100; i++) {
```
Comment (Max-Pochepets) [inline src/main/java/MyRunnable.java:14]: You have Counter and it's value, no need for another variable

File: src/main/java/MyThread.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    @Override
+    public void run() {
+        logger.info("MyThread thread started:" + System.lineSeparator());
+        for (int i = 0; i < 100; i++) {
+            logger.info(Thread.currentThread().getName() + " value = "
+                    + counter.increment() + System.lineSeparator());
+        }
+        logger.info("MyThread finished." + System.lineSeparator());
+    }
+}
```
Comment (Max-Pochepets) [inline src/main/java/MyThread.java:20]: Is there an empty line?

**Review by Max-Pochepets — CHANGES_REQUESTED** [review]

**Review by Max-Pochepets — APPROVED** [review]

File: src/main/java/Counter.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+public class Counter {
+    private static int value = 0;
```
Comment (Sofasmile) [inline src/main/java/Counter.java:2]:
```suggestion
    private int value = 0;
```

File: src/main/java/MyThread.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import org.apache.log4j.Logger;
+
+public class MyThread extends Thread {
+    private static final Logger logger = Logger.getLogger(MyThread.class);
+    private Counter counter;
+    int maxValue;
```
Comment (Sofasmile) [inline src/main/java/MyThread.java:6]:
```suggestion
    private int maxValue;
```

File: src/main/java/MyThread.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        this.counter = counter;
+        this.maxValue = maxValue;
+    }
+
+    @Override
+    public void run() {
+        logger.info("MyThread thread started:" + System.lineSeparator());
+        while (counter.getValue() < maxValue) {
+            logger.info(Thread.currentThread().getName() + " value = "
+                    + +counter.getValue() + System.lineSeparator());
```
Comment (Sofasmile) [inline src/main/java/MyThread.java:18]:
```suggestion
                    + counter.getValue() + System.lineSeparator());
```

File: src/main/resources/log4j.properties [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+log4j.appender.stdout.Target=System.out
+log4j.appender.stdout.layout=org.apache.log4j.PatternLayout
+log4j.appender.stdout.layout.ConversionPattern=%d{ABSOLUTE} %-5p %c{1}:%L - %m%n
+# Redirect log messages to a log file, support file rolling.
+log4j.appender.file=org.apache.log4j.RollingFileAppender
+log4j.appender.file.File=/home/alex/IdeaProjects/multithreading/logs/app.log
+log4j.appender.file.MaxFileSize=5MB
+log4j.appender.file.MaxBackupIndex=10
+log4j.appender.file.layout=org.apache.log4j.PatternLayout
+log4j.appender.file.layout.ConversionPattern=%d{ABSOLUTE} %-5p %c{1}:%L - %m%n
```
Comment (Sofasmile) [inline src/main/resources/log4j.properties:14]: add empty line

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/MyThread.java (hunk 4 above)
Comment (Sofasmile) [inline src/main/java/MyThread.java:6]: не бачу змін

**Review by Sofasmile — CHANGES_REQUESTED** [review]

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**